src/NHibernate/Cache/HashtableCacheProvider.cs

```diff
@@ -10,7 +11,14 @@ public class HashtableCacheProvider : ICacheProvider
 	{
 		#region ICacheProvider Members
 
-		public ICache BuildCache(string regionName, IDictionary<string, string> properties)
+		// Since 5.2
+		[Obsolete]
+		ICache ICacheProvider.BuildCache(string regionName, IDictionary<string, string> properties)
+		{
+			return BuildCache(regionName, properties);
+		}
+
+		public CacheBase BuildCache(string regionName, IDictionary<string, string> properties)
```

Comment:
This implementation of `BuildCache` allows to get the provider almost ready for the switch from `ICache` to `CacheBase`. It will only have to drop the explicit implementation.